### ISSUE TITLE
Preserve @NonNull on type variables through wildcard capture

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1823]
NullAway reports false positives when an explicitly `@NonNull` type variable is accessed through a wildcard capture. This affects both a direct return type (`@NonNull V`) and a nested type argument (`ConcurrentMap<String, @NonNull V>`).

Originally reported by @ben-manes in [Caffeine issue #2004, finding 1](https://github.com/ben-manes/caffeine/issues/2004#issuecomment-5467003874). Caffeine encountered this when reading entry values from `Cache<?, ?>.asMap()`.

### Reproducer

Run with `JSpecifyMode=true` and `JSpecifyExperimental=true`:

```java
import java.util.concurrent.ConcurrentMap;
import org.jspecify.annotations.*;

@NullMarked
class Test {
  interface Holder<V extends @Nullable Object> {
    @NonNull V value();
    ConcurrentMap<String, @NonNull V> asMap();
  }

  static void takesNonNull(Object value) {}

  static void explicitNullableArgument(Holder<@Nullable Object> holder) {
    takesNonNull(holder.value()); // accepted
    holder.asMap().entrySet().forEach(e -> takesNonNull(e.getValue())); // accepted
  }

  static void nonNullBoundedWildcard(Holder<? extends Object> holder) {
    takesNonNull(holder.value()); // accepted
    holder.asMap().entrySet().forEach(e -> takesNonNull(e.getValue())); // accepted
  }

  static void unboundedWildcard(Holder<?> holder) {
    takesNonNull(holder.value()); // false positive
    holder.asMap().entrySet().forEach(e -> takesNonNull(e.getValue())); // false positive
  }

  static void nullableBoundedWildcard(Holder<? extends @Nullable Object> holder) {
    takesNonNull(holder.value()); // false positive
    holder.asMap().entrySet().forEach(e -> takesNonNull(e.getValue())); // false positive
  }
}
```

### Expected behavior

All eight calls to `takesNonNull` should be accepted. The explicit `@NonNull V` excludes null regardless of the type argument or captured wildcard's upper bound.

### Actual behavior

Each of the last two methods produces both diagnostics:

```text
[NullAway] passing @Nullable parameter 'holder.value()' where @NonNull is required
[NullAway] passing @Nullable parameter 'e.getValue()' where @NonNull is required
```

The explicit nullable type argument and non-null-bounded wildcard controls are accepted.

**Comment by KaustAbhinand** [comment]
Hi @msridhar! Could I work on this issue? I'll submit a PR soon :)

**Comment by msridhar** [comment]
Hi @KaustAbhinand I'm already working on this one

**Comment by KaustAbhinand** [comment]
Are there any other issues which I could take up? I am happy to contribute.

**Comment by msridhar** [comment]
@KaustAbhinand for ease can you hop on the NullAway Discord?

https://discord.gg/QH2F779

We can discuss there.
